store/derivatives.ts

```diff
@@ -859,15 +862,32 @@ export const actions = actionTree(
       await this.app.$accessor.app.queue()
       await this.app.$accessor.wallet.validate()
 
-      await closePosition({
-        quantity,
-        price,
-        injectiveAddress,
-        address,
-        orderType,
-        market: (currentMarket || market) as UiDerivativeMarket,
-        subaccountId: subaccount.subaccountId
-      })
+      if (reduceOnlyOrders.length) {
+        await closePositionAndReduceOnlyOrders({
+          quantity,
+          price,
+          injectiveAddress,
+          address,
+          orderType,
+          market: (currentMarket || market) as UiDerivativeMarket,
+          subaccountId: subaccount.subaccountId,
+          reduceOnlyOrders: reduceOnlyOrders.map((o) => ({
+            orderHash: o.orderHash,
+            subaccountId: o.subaccountId,
+            marketId: o.marketId
+          }))
+        })
+      } else {
```

Comment:
As a general rule in our codebase we try to avoid `if -> else` statements and utilize early returns, you can read more about it here: https://codeburst.io/code-js-in-stlye-early-return-3462323e30d1

Reply:
"If a condition is true 
  return value."
^^ I like it!